onnxruntime/contrib_ops/rocm/bert/tunable_op.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <iostream>
+#include <map>
+#include <vector>
+#include <string>
+#include "timer.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace rocm {
+
+struct OpParams {
+  virtual std::string signature() const = 0;
+};
+
+class Op {
```

Comment:
it looks more like a profiler, instead of an op :)

Reply:
Yeah, that's true, profiling is the internal mechanism to pick the fastest implementation. I still prefer to call this an op though, as to a user, this is still very much an op (kind of a smarter one that can make some internal choices) and a user doesn't need to care or know much about the internal op selection method (whether it is profiling-based, modeling-based, or learning-based, and so on).